### PR TITLE
Remove Dup AES-GCM Code and Split Off OpenSSL code

### PR DESCRIPTION
**Pull request by danintel**
This splits file skenc.cpp into OpenSSL-dependent and -independent parts.
It also removes multiple implementations of EncryptMessage().

- skenc.cpp
  - keep low-level OpenSSL-dependent code in this file
  - EncryptMessage(): remove duplicate implementations

- skenc_common.cpp (new file)
  - Move OpenSSL-independent code from skenc.cpp to here
  - GenerateIV(): use Avalon hash function instead of OpenSSL hash function

- Makefile
  - Clean up duplicate rules

This will make implementing skenc.cpp with MbedTLS easier.

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

File: common/cpp/crypto/skenc.cpp [hunk 1]
```diff
@@ -16,70 +16,38 @@
...
 /**
  * Encrypt a message using AES-GCM authenticated encryption.
  *
- * Prepends a 12 byte (96 bit) IV (sometimes called a nonce) and
- * appends a 16 byte (128 bit) authentication tag (sometimes called a MAC)
+ * Appends a 16 byte (128 bit) authentication tag (sometimes called a MAC)
  * to the output cipher text:
- *     message = IV + ciphertext + authentication tag
- * The IV and nonce are not encrypted.
+ *     message = ciphertext + authentication tag
```
[the remark this answers is not in the document]
Comment (danintel) [reply common/cpp/crypto/skenc.cpp:48]: @manojgop The comment is wrong. I added the incorrect comment a few PRs ago. I did not change the interfaces, but I fixed the incorrect comment I previously added.

Basically, there are two flavors of `EncryptMessage()`:

- EncryptMessage() where the IV is passed as a parameter (this case)
- EncryptMessage() where the IV is prepended to the message.

[the remark this answers is not in the document]
Comment (danintel) [reply common/cpp/crypto/skenc.cpp:48]: @manojgop Whether the IV is a parameter or prepended to the message is implementation dependent. NIST specifies the algorithm, but not the code or data structures.

[the remark this answers is not in the document]
Comment (danintel) [reply common/cpp/crypto/skenc.cpp:48]: @manojgop: I see the confusion now.  I removed the "IV prepended" comment in the code.

[the remark this answers is not in the document]
Comment (manojgop) [reply common/cpp/crypto/skenc.cpp:48]: @danintel ok. PLease note that Avalon by default uses EncryptMessage API which by defaults genarates a random IV and prepends 96 bit IV in encrypted message. Similarly DecryptMessage also assumes that 96bit IV is added in encrypted data. https://github.com/hyperledger/avalon/blob/master/common/cpp/crypto/skenc_common.cpp#L97

[the remark this answers is not in the document]
Comment (danintel) [reply common/cpp/crypto/skenc.cpp:48]: @manojgop The higher-level functions uses a pre-pended IV, if desired, and that is still available.

**Pull request merged**